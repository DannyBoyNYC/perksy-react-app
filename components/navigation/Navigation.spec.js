const React = require('react');
const { mount, shallow } = require('enzyme');

const Navigation = require('./index').default;

describe('<Navigation.* />', () => {
  let component;
  const requiredProps = {
    links: [
      {
        text: 'Foo',
        url: '#'
      },
      {
        text: 'Bar',
        url: '#'
      }
    ]
  };

  describe('<Navigation.Horizontal />', () => {
    beforeEach(() => {
      component = shallow(<Navigation.Horizontal {...requiredProps} />).dive();
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should have the correct BEM class names', () => {
      expect(component).toHaveClassName('navigation navigation--horizontal');
    });

    it('will render icons when passed', () => {
      const linksWithIcons = [
        {
          icon: <i className="fas fa-check" />,
          text: 'Done',
          url: '#'
        },
        {
          icon: <i className="far fa-square" />,
          text: 'To Do',
          url: '#'
        }
      ];
      component = mount(<Navigation.Horizontal links={linksWithIcons} />);
      expect(component).toContainMatchingElements(2, '.navigation-item__icon');
    });
  });

  describe('<Navigation.Vertical />', () => {
    beforeEach(() => {
      component = shallow(<Navigation.Vertical {...requiredProps} />).dive();
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should have the correct BEM class names', () => {
      expect(component).toHaveClassName('navigation navigation--vertical');
    });
  });
});
