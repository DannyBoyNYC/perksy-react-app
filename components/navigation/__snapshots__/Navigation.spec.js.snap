// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation.* /> <Navigation.Horizontal /> should render correctly with all required props 1`] = `
<nav
  className="navigation navigation--horizontal"
>
  <Unordered
    className="navigation__list"
    items={
      Array [
        Object {
          "text": "Foo",
          "url": "#",
        },
        Object {
          "text": "Bar",
          "url": "#",
        },
      ]
    }
    onPrerender={[Function]}
  />
</nav>
`;

exports[`<Navigation.* /> <Navigation.Vertical /> should render correctly with all required props 1`] = `
<nav
  className="navigation navigation--vertical"
>
  <Unordered
    className="navigation__list"
    items={
      Array [
        Object {
          "text": "Foo",
          "url": "#",
        },
        Object {
          "text": "Bar",
          "url": "#",
        },
      ]
    }
    onPrerender={[Function]}
  />
</nav>
`;
